File: practice_datastructure/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;
        for(int i = 0; i < nums.size(); i++){
            int remain = target - nums[i];
            if(m.count(remain) != 0){
                return {m[remain], i};
            }else{
                m[nums[i]] = i;
            }
        }
        return {};
    }
};

File: PastYears/practice_datastructure/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer>m = new HashMap<>();
        for(int i = 0;  i < nums.length; i++){
            int remain = target - nums[i];
            if(m.containsKey(remain)){
                int pos = m.get(remain);
                return new int[]{pos, i};
            }else{
                m.put(nums[i], i);
            }
        }
        return new int[2];
    }
}
